Add information about CloudEvents and improve the Trigger tutorial (#8759)
kyma-project/kyma: docs/event-mesh/03-01-event-processing.md

    >**NOTE:** The HTTP Source Adapter accepts only CloudEvents in version 1.0.

>   Before Kyma Version 1.11, the application sent events to the `/v1/events` endpoint exposed by the [Event Service](https://github.com/kyma-project/kyma/blob/master/components/event-service/README.md). This service was responsible for sending events to the Event Bus, which back then handled event processing and delivery. In Event Mesh, the application sends events to the directly exposed `/events` endpoint. These events must comply with the [CloudEvents specification](https://github.com/cloudevents/spec/blob/v1.0/spec.md).

Reviewer: ```suggestion
  Before Kyma 1.11, applications sent events to the `/v1/events` endpoint exposed by the [Event Service](https://github.com/kyma-project/kyma/blob/master/components/event-service/README.md). This service was responsible for sending events to the Event Bus that used to handle event processing and delivery. In the current implementation based on the Event Mesh, applications send events to the directly exposed `/events` endpoint. These events must comply with the [CloudEvents specification](https://github.com/cloudevents/spec/blob/v1.0/spec.md).
```

>   Before Kyma Version 1.11, applications sent events to the `/v1/events` endpoint exposed by the [Event Service](https://github.com/kyma-project/kyma/blob/master/components/event-service/README.md). This service was responsible for forwarding events to the Event Bus, that used to handle event processing and delivery. In the current implementation based on the Event Mesh, applications send events to the directly exposed `/events` endpoint. These events must comply with the [CloudEvents specification](https://github.com/cloudevents/spec/blob/v1.0/spec.md).

Reviewer: `Event Bus, that used to handle event processing and delivery` -> I think this comma is not needed here - `that` refers directly to Event Bus that precedes it.

Reviewer: `Before Kyma Version 1.11` -> Before Kyma 1.11

>   It may happen, that you still use the previous endpoint to receive the events. That is why the Event Service's logic was refactored into a compatibility layer to ensure that the events received at the `/v1/events` endpoint can be forwarded to the subscriber. To achieve that, it consumes the events in the non-CloudEvent format, transforms them into CloudEvents, and propagates to the new Event Mesh. 

Reviewer: ```suggestion
  It may happen that you still use the previous endpoint to receive events. That is why we equipped the Event Service with an additional compatibility layer that ensures the events sent to the `/v1/events` endpoint can be forwarded to the subscriber. To achieve that, the compatibility layer consumes the events in the non-CloudEvent format, transforms them into CloudEvents, and propagates them to the new Event Mesh. 
```

>   It may happen, that you still use the `/v1/events` endpoint to receive events. To support it, we extended the Event Service's logic to ensure all events sent to this endpoint are forwarded to the subscriber. This way, the Event Service acts as a compatibility layer that consumes the events in the non-CloudEvent format, transforms them into CloudEvents, and propagates them to the new Event Mesh. 

Reviewer: ```suggestion
  It may happen that you still use the `/v1/events` endpoint to receive events. To support it, we extended the Event Service's logic to ensure all events sent to this endpoint are forwarded to the subscriber. This way, the Event Service acts as a compatibility layer that consumes the events in the non-CloudEvent format, transforms them into CloudEvents, and propagates them to the new Event Mesh. 
```

>   It may happen, that you still use the previous endpoint to receive the events. That is why we extended the Event Service's logic to ensure the events sent to the `/v1/events` endpoint are forwarded to the subscriber. This way, the Event Service acts as a compatibility layer that consumes the events in the non-CloudEvent format, transforms them into CloudEvents, and propagates them to the new Event Mesh. 

Reviewer: ```suggestion
  It may happen, that you still use the `/v1/events` endpoint to receive events. To support it, we extended the Event Service's logic to ensure all events sent to the `/v1/events` endpoint are forwarded to the subscriber. This way, the Event Service acts as a compatibility layer that consumes the events in the non-CloudEvent format, transforms them into CloudEvents, and propagates them to the new Event Mesh. 
```
